environments/openstack/README.md

```diff
@@ -1,17 +1,135 @@
+
+**Important**: Head to the `main.tf` file to see which of them are configurable. Keep in mind that
+`image_id`, `flavor_id` and `external_net` properties will be different across
+OpenStack instances. So you need to set them **manually**!
+
+
+To start with the provisioning we firstly need to download 
+the [`Openstack RC file`](https://docs.openstack.org/zh_CN/user-guide/common/cli-set-environment-variables-using-openstack-rc.html) 
+from your Openstack instance. To do so proceed to the **Computations - Access and Security - API access** tab and
+click on **Download OpenStack RC File v3**.
+
+This file hold all needed variables needed for successful authorization
+to the Openstack during provisioning.
+
+To apply those variables source the file (you will be prompted for password
+to your OpenStack instance):
 ```bash
 $ source meddtager-project.sh
 ```
 
-Finally:
+Now you are all set and you can start provisioning, to do so simply run:
+
 ```bash
 $ make deploy
 ```
 
-**Tip:** You can use `terraform plan` to see what changes would be done, without actually apllying them.
+**Tip:** You can use `terraform plan` to see what changes would be done, 
+without actually applying them.
+
+### Ansible deployment
+
+**Note**: It is assumed that:
+- you have already completed the Terraform part of this deployment.
+If not, you will have to create the `inventory` and `backend.env` files manually. 
+Head to terraform folder and look at the `inventory.tpl` and `backend.env.tpl` files to 
+learn how does their look like,
+- the 2 private keys are placed in the Ansible folder.
+
+After this step you will have:
+- MedTagger source code on both of machines,
+- Private key to DbHost copied to AppHost,
+- Docker and Docker-compose installed on both of machines,
+- `backend.env` and `docker-compose.yml` swapped (head to the next
+section to learn what changed)
+
+Before we run this script head to the `group_vars/all.yml` file to 
+make sure that the variables are set properly!
+
+To install all dependencies and start the docker containers run (from the Ansible directory): 
+```bash
+$ ansible-playbook site.yml -i inventory
+```
+
+**Tip**: You will be prompted to accept the ssh key. You can disable it by 
+adding `ANSIBLE_HOST_KEY_CHECKING=False ansible-playbook ...`
+
+### Configuration for Docker Swarm
+
+Since we are setting up MedTagger using Docker Swarm, there were few changes that were needed
+in our default docker-compose.yml.
+
+1. Placement of the services.
+
+    As you can see in the file, all of the services now how their placement
+    directly specified with the following configuration:
+    
+    ```    
+    deploy:
+          placement:
+            constraints:
+              - node.hostname == apphost
+    ```
+
+2. Enabling Docker Swarm mode in Traefik.
+
+    To be able to use Traefik we need to enable Docker Swarm mode. You can
+    read more about it [here](https://docs.traefik.io/user-guide/cluster-docker-consul/).
```

Comment:
You attached wrong tutorial, don't you? :)
https://docs.traefik.io/user-guide/swarm-mode/